crates/rattler_installs_packages/src/resolve/dependency_provider.rs

```diff
@@ -213,6 +216,16 @@ impl PypiDependencyProvider {
             .iter()
             .any(|a| a.is::<S>())
     }
+
+    /// Acquires a lease to be able to spawn a task
+    /// this is used to limit the amount of concurrent tasks
+    async fn aquire_lease_to_run(&self) -> tokio::sync::OwnedSemaphorePermit {
+        self.concurrent_tasks
+            .clone()
+            .acquire_owned()
+            .await
+            .expect("could not acquire semaphore")
```

Comment:
It is ok to fail here with expect or we need to have later a mechanism for recovering from it?

Reply:
I check it and it should only fail when the semaphore is closed, which we should never call.